Add pipeline setup to index and create operations

renatogm24 opened the pull request:
This pull request introduces a change to allow the pipeline in index operations to be set up individually. Previously, the pipeline was not set up for each index operation, which resulted in using the default pipeline on the index.

The changes include:
- Adding a copyIndexProperties method to copy the pipeline property from the IndexOperation to the IndexOperation.Builder.
- Modifying the indexOperation method to use copyIndexProperties instead of copyBaseProperties, ensuring the pipeline is copied for each index operation.
- Similar changes have been made for createOperation method to ensure consistency.

These changes allow each request to have its corresponding pipeline, providing more flexibility and control over how documents are indexed. This is particularly useful when different documents in the same index require different processing.

This change aligns with the existing structure of the BulkRequest, which also includes a "pipeline" field. With this change, the pipeline field in the BulkRequest will be correctly set up by the BulkIngester.

swallez commented on java-client/src/main/java/co/elastic/clients/elasticsearch/_helpers/bulk/IngesterOperation.java:
```diff
@@ -154,6 +154,16 @@ private static void copyBaseProperties(BulkOperationBase op, BulkOperationBase.A
 
+    private static void copyIndexProperties(IndexOperation<?> op, IndexOperation.Builder<?> builder) {
+        copyBaseProperties(op, builder);
+        builder.pipeline(op.pipeline());
```
Can you please add `requireAlias` that is also part of index operations?

```suggestion
        builder.pipeline(op.pipeline());
        builder.requireAlias(op.requireAlias());
```

renatogm24 replied:
Yes sure, it is done.

swallez reviewed (changes requested):
Thanks @renatogm24. This is actually a bug, thanks for the catch and the fix!

For the fix to be complete, I added suggestions for `requireAlias` which is also missing.

Would you mind also adding this test in `BulkIngesterTest`?

```java
    @Test
    public void pipelineTest() {
        String json = "{\"create\":{\"_id\":\"some_id\",\"_index\":\"some_idx\",\"pipeline\":\"pipe\",\"require_alias\":true}}";
        JsonpMapper mapper = new SimpleJsonpMapper();

        BulkOperation create = BulkOperation.of(o -> o.create(c -> c
            .pipeline("pipe")
            .requireAlias(true)
            .index("some_idx")
            .id("some_id")
            .document("Some doc")
        ));

        String createStr = JsonpUtils.toJsonString(create, mapper);
        assertEquals(json, createStr);

        BulkOperation create1 = IngesterOperation.of(create, mapper).operation();

        String create1Str = JsonpUtils.toJsonString(create1, mapper);
        assertEquals(json, create1Str);
    }
```

Once done, I'll be happy to merge this PR.

Fixes #586

renatogm24 commented on java-client/src/main/java/co/elastic/clients/elasticsearch/_helpers/bulk/IngesterOperation.java:
```diff
@@ -154,6 +154,16 @@ private static void copyBaseProperties(BulkOperationBase op, BulkOperationBase.A
+
+    private static void copyCreateProperties(CreateOperation<?> op, CreateOperation.Builder<?> builder) {
+        copyBaseProperties(op, builder);
+        builder.pipeline(op.pipeline());
```
Yes sure, it is done.

renatogm24 commented:
Hello @swallez, thanks for your response. I have just added the requireAlias to both copyProperties methods and also added the pipelineTest that you suggested, which is working. Thanks for your help.

swallez reviewed (approved):
LGTM. Thanks for the fix!
